specification/loadtestservice/LoadTestService/models.tsp

```diff
@@ -498,7 +498,7 @@ model TestInputArtifacts {
 model FileInfo {
   @doc("Name of the file.")
   @key
-  @visibility
+  @visibility("read")
```

Comment:
I can also remove the `@visibility` (and do a suppression)

But this `@visibility` without parameter would cause TCGC output a model with empty name.
![image](https://github.com/Azure/azure-rest-api-specs/assets/53292327/c3fe682b-6420-4ca5-bc65-2480424b8e83)

Reply:
oh wait, you already notice that.

Reply:
> does remove `@visibility` here give us the same result? I remember `@key` will set the property as readOnly in the response?

I think `@key` along won't.
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/ContentSafety/models.tsp#L147-L148

Java emitter actually takes that `isSegment` means read-only. But TCGC won't.

Reply:
This is a response-only model -- never used for input -- is that correct? If so then then best fix at this point is to just remove the `@visibility` decorator altogether. `readOnly` is meaningless for properties never sent in input, and removing `@visibility` completely will eliminate the breaking change.

Reply:
@mikekistler 

Yeah, if we look at Swagger, this probably the good approach (Swagger generally do not want `required` + `readOnly`).

The reason here is that typespec would give a warning if no `@visibility` on `@key`. And I believe this is the cause that this `@visibility` existed.
([playground](https://cadlplayground.z22.web.core.windows.net/cadl-azure/?c=aW1wb3J0ICJAdHlwZXNwZWMvdmVyc2lvbmluZyI7CskfYXp1cmUtdG9vbHMvyCstxhVjb3JlIjsKCnVzaW5nIFR5cGVTcGVjLlJlc3Q70BVIdHRw0RVWyXLIG0HEWS5Db3JlLlRyYWl0czsKCkBzZXJ2aWNlKHt0aXRsZTogIlRleHQgYXV0aG9y5ACufSkKQOcAvWVkKFPGLsdccykKbmFtZXNwYWNlIMQ4Qcg3OwoKZW51bSDPMCB7CiAgQHVzZURlcGVuZGVuY3ko6wCWyCcudjFfMF9QcmV2aWV3XzIpCiAgdjE6ICIyMDIxLTDEAyIsCn0KCmludGVyZsR9UmVzb3VyY2VPcGVyYXTERAogIGV4dGVuZHPsAPjSKDxOb1JlcGVhdGFibGVSZXF1ZXN0cyAmIE5vQ29uZGnEJGFszhhsaWVudMcTSWQ%2BIHt9CgovKioKICogZG9jCiAqLwpAcsdpKCJwcm9qZWN0cyIpCm1vZGVsIFDGEeUBDMU3IMg5IMQ7ICBAa2V5xBwvICNzdXBwcmVzc%2F8CLG9yZS9rZXktdmlzaWJpbGl0eS1yZXF1aXJlZCIKICDkAaU6IHN05gGa7QE56QGy5QCPbGlzdCBpc%2FMBT%2BkBOlJlYWQ85wDDPsRN&e=%40azure-tools%2Ftypespec-autorest&options=%7B%22linterRuleSet%22%3A%7B%22extends%22%3A%5B%22%40azure-tools%2Ftypespec-azure-rulesets%2Fdata-plane%22%5D%7D%7D))

We can do a 
```
#suppress "@azure-tools/typespec-azure-core/key-visibility-required"
```
If you think this be better solution.

Reply:
Best to check with the TypeSpec folks on their linter rule, but unless they have a strong rationale for requiring @visibility then I would add the suppression.

Reply:
This sounds good to me, currently there are no plans to refactor and use it as input model as well

Reply:
Sure. Let's do a `suppress` then.

There is [long discussion in channel](https://teams.microsoft.com/l/message/19:906c1efbbec54dc8949ac736633e6bdf@thread.skype/1715929611117?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1715929611117&teamName=Azure%20SDK&channelName=TypeSpec%20Discussion&createdTime=1715929611117) about the combinations of `@key`, `@visibility`, `@path` (and difference of their effect with `@body` or `@bodyRoot`)...